File: expressions/budget.c
#include <stdio.h>

int main(void){
    float income, rent, utilities, groceries, transportation, expenses, savings, total;
    float prent, putilities, ptransportation, pgroceries, pexpenses;
    printf("This is going to calculate your budget for the month\n");
    printf("how much do you make a month?\n");
    scanf("%f", &income);

    printf("how much is rent?\n");
    scanf("%f", &rent);   
    printf("how much is utilities?\n");
    scanf("%f", &utilities);  
    printf("how much is groceries?\n");
    scanf("%f", &groceries); 
    printf("how much is transportation?\n");
    scanf("%f", &transportation);   
    expenses = rent + utilities + groceries + transportation;
    savings = income *.2;
    total = income - expenses - savings;
    prent = rent/income;
    putilities = utilities/income;
    pgroceries = groceries/income;
    ptransportation = transportation/income;
    pexpenses = expenses/income;

    printf("your income is: $%.2f\n", income);
    printf("Your expenses: $%.2f\n",expenses); 
    printf("Your total left to spend $%.2f\n", total);
    printf("your rent is %.2f", prent, "of your income");
    printf("your utilities is %.2f", putilities, "of your income");

 return 0;   
}

File: expressions/budget.py

print("Hello, and welcome to your finacial calculator!\n")
income = input ("how much do you make a month: \n")
utilities = input ("how much is your monthly utilities: \n")
rent = input ("how much is your monthly house or rent payment: \n")
groceries = input ("how much is your monthly groceries: \n")
transportation = input ("how much is your monthly transportation: \n")

expenses = rent + utilities + groceries + transportation
savings = income * 0.2
total = income - expenses - savings
prent = rent/income *100
putilities = utilities/income *100
pgroceries = groceries/income *100
ptransport = transportation/income *100
pexpenses = expenses/income *100

print("percent of your rent",prent)
print("percent of your utilities",putilities)
print("percent of your groceries",pgroceries)
print("percent of your transport",ptransport)
print("percent of your expenses",pexpenses)